### PR TITLE
update logger with dynamic function name

### PR DESCRIPTION
**Pull request by ducquangkstn**

### CODE REVIEW

**Comment by ducquangkstn** [comment]
This is just refactor PR, but it changes a lot of files. Reviewing this task should be the low priority

File: lib/caller/function_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetCallerFunctionName(t *testing.T) {
+	names := firstFunction()
+	require.Equal(t, len(names), 3)
+	assert.Equal(t, "github.com/KyberNetwork/reserve-stats/lib/caller.firstFunction", names[0])
```
Comment (secmask) [inline lib/caller/function_test.go]: I would recommend to strip `{domain}/{organize}/` from function name, we just need to keep `reserve-stats/lib/caller.firstFunction`, so we can make a copy of `GetCurrentFunctionName` like `GetCurrentFunctionNameShort(2)` which will strip 2 first components in func name.

**Comment by secmask** [comment]
@halink0803 this may affect if we search func name in Kibana

File: lib/caller/function_test.go (hunk 1 above)
Comment (ducquangkstn) [reply lib/caller/function_test.go]: I have added a function to remove 2 first components in func name if func name has more than 2 components

**Pull request merged**